File: langs/group-pine/dart/def/sample.dart
class Person {
  final String name;
  final int age;

  Person(this.name, this.age);

  String greet() => 'Hello, my name is $name';
}

void main() {
  final person = Person('Alice', 30);
  print(person.greet());

  final numbers = [1, 2, 3, 4, 5];
  final doubled = numbers.map((n) => n * 2).toList();
  print(doubled);
}

File: crates/group-a/arborium-rust/sample.rs
use std::fmt;

pub trait Greet {
    fn greet(&self) -> String;
}

#[derive(Debug, Clone)]
pub struct Person {
    name: String,
    role: Role,
}

pub enum Role {
    Admin,
    User { permissions: Vec<String> },
}

impl Greet for Person {
    fn greet(&self) -> String {
        format!("Hello, I'm {}!", self.name)
    }
}

impl fmt::Display for Person {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{} ({:?})", self.name, self.role)
    }
}

fn main() {
    let alice = Person {
        name: "Alice".into(),
        role: Role::Admin,
    };
    println!("{}", alice.greet());
}
